eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/StashPass.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * File containing the StashPass class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler;
+
+use eZ\Bundle\EzPublishCoreBundle\Cache\Driver\Redis\RedisIgbinary;
+use eZ\Bundle\EzPublishCoreBundle\Cache\Driver\Redis\RedisIgbinaryLzf;
+use eZ\Bundle\EzPublishCoreBundle\Cache\Driver\Redis\RedisSerializeLzf;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * This compiler pass overrides default Stash's Redis driver with one of stored in eZ/Bundle/EzPublishCoreBundle/Cache/Driver.
+ */
+class StashPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('stash.driver')) {
+            return;
+        }
+
+        $igbinary = $container->hasParameter('ezpublish.stash_cache.igbinary') ? $container->getParameter('ezpublish.stash_cache.igbinary') : false;
+        $lzf = $container->hasParameter('ezpublish.stash_cache.lzf') ? $container->getParameter('ezpublish.stash_cache.lzf') : false;
+
+        $config = $container->getExtensionConfig('stash');
+        $config = reset($config);
+        foreach ($config['caches'] as $name => $configuration) {
+            if (in_array('Redis', $configuration['drivers'], true)) {
+                $this->configureRedis($container, $igbinary, $lzf);
+            }
+        }
+
+        $stashDriverDef = $container->findDefinition('stash.driver');
+        $stashDriverDef->setFactory(
```

Comment:
Yes, it works. Thank you for drawing my attention on this. Changed in https://github.com/ezsystems/ezpublish-kernel/pull/2387/commits/692446f17df97964a7b24c7dd2b44b896b009b64 as well as other changes which you've requested.

Reply:
@alongosz I have to go back to the `DriverFactory` since what you've proposed indeed works but only in `dev` mode when most of the container parts are refreshed on each request. 

In `prod` mode, this `DriverList::registerDriver($name, $class);` is executed *only once* on container build, but besides that, driver is registered on each request. Therefore, one of *our* drivers indeed is registered instead of default Stash driver, but only during the container compilation. On every next request this static `DriverList::getAllDrivers` is used again, but without changing its content, so we end-up with default Stash's Redis driver.